src/core/comm/comm_nccl.cu

```diff
@@ -170,6 +189,20 @@ void register_tasks(Legion::Machine machine,
   }
 }
 
+static const char* MIN_VERSION_WITH_HANG_FIX = "520";
+
+bool needs_barrier()
+{
+  CHECK_NVML(nvmlInit_v2());
```

Comment:
Could you add a couple of sentences here for future readers, describing the issue and workaround? It's clear by reading the code how the workaround functions, but it's not clear why it's needed.

Reply:
72cd10eb3abbfcdde9d6d0d8faab5289592e4764